### PR TITLE
EDGECLOUD-3143: [Unity Pong] Location is Missing on Device

### PR DESCRIPTION
**Pull request by franklin-huang-mobiledgex**
Missing InitializeLocationService function call and in order to use that function, we have to call it from the main thread.

1. Start() is now a unity coroutine instead of async Task
2. Add CheckLocationIsRunning() coroutine that waits until LocationService is running before calling MobiledgeXAPICalls 
(@lgarner I think we should add a similar function to Unity SDK Location Service class. Maybe it takes a callback/delegate as a parameter and we wait until LocationService is running before calling the callback)
3. Remove PingPong LocationService script

### CODE REVIEW

File: unity/PingPongGameExample/PingPongGame/Assets/Scripts/GameManager.cs [hunk 1]
```diff
@@ -140,7 +140,30 @@ async Task Start()
       roomIdInput = GameObject.Find("InputFieldRoomId").GetComponent<InputField>();
       roomIdInput.onEndEdit.AddListener(ConnectToServerWithRoomId);
 
+      yield return CheckLocationIsRunning();
```
Comment (ah1053) [inline unity/PingPongGameExample/PingPongGame/Assets/Scripts/GameManager.cs:143]: yield return StartCoroutine(CheckLocationIsRunning());
will start the Coroutine before the yield and wait for it, I think this what u intending to do

Comment (franklin-huang-mobiledgex) [reply unity/PingPongGameExample/PingPongGame/Assets/Scripts/GameManager.cs:143]: What is the difference in behavior between StartCoroutine(CheckLocationIsRunning()); and CheckLocationIsRunning();?

File: unity/PingPongGameExample/PingPongGame/Assets/Scripts/GameManager.cs [hunk 2]
```diff
@@ -140,7 +140,30 @@ async Task Start()
...
+
+    IEnumerator CheckLocationIsRunning()
+    {
       MobiledgeX.LocationService.ensurePermissions();
+      StartCoroutine(MobiledgeX.LocationService.InitalizeLocationService());
+
+#if UNITY_EDITOR
+      if (Input.location.status == LocationServiceStatus.Failed)
+      {
+        clog("Location service failed for some reason");
```
Comment (vmohanx) [inline unity/PingPongGameExample/PingPongGame/Assets/Scripts/GameManager.cs:154]: Can we update this message to say something like "Location service is disabled in Unity Editor"

Comment (franklin-huang-mobiledgex) [reply unity/PingPongGameExample/PingPongGame/Assets/Scripts/GameManager.cs:154]: I can change this to something like "Location service failed". But actually in Editor, Location Service is not disabled (LocationServiceStatus.Stopped). LocationServiceStatus.Failed would be a critical error

Comment (vmohanx) [reply unity/PingPongGameExample/PingPongGame/Assets/Scripts/GameManager.cs:154]: sure. mainly I would want to add the fact the error is being printed from the editor path. so "Location service failed in Unity Editor"

File: unity/PingPongGameExample/PingPongGame/Assets/Scripts/GameManager.cs (hunk 1 above)
Comment (ah1053) [reply unity/PingPongGameExample/PingPongGame/Assets/Scripts/GameManager.cs:143]: The difference is one frame , 

yield return CheckLocationIsRunning(); > yield return null (one frame) >> StartCoroutine(CheckLocationIsRunning()) >>  continue

yield return StartCoroutine(CheckLocationIsRunning()) > StartCoroutine(CheckLocationIsRunning()) >> continue

Screen Shot 2020-07-06 at 10 57 37 AM
[link](https://docs.unity3d.com/ScriptReference/MonoBehaviour.StartCoroutine.html)

**Pull request merged**